Tasks/AzureCLIV2/azureclitask.ts

```diff
@@ -54,10 +54,13 @@ export class azureclitask {
             }
 
 
-            if (failOnStdErr) {
+            if (failOnStdErr && aggregatedErrorLines.length > 0) {
+                let error = "";
                 aggregatedErrorLines.forEach((err: string) => {
                     tl.error(err);
+                    error += err;
                 });
+                throw error;
```

Comment:
Having said that, there definitely is the problem when exit code is 0 but things are pushed to std err. So this line is needed. But we do not need to throw the error message. The message has already been printed using t.error() and hence we should just throw a very simple error message like "script pushed messages in standard output and hence failing the task"